### PR TITLE
Glob include files based on project extension (VS)

### PR DESCRIPTION
**Pull request by ViktorHofer**
Wasn't reported but something that I just noticed while scanning code. Extracting into the root Directory.Build.targets, as this is the behavior that we want to see throughout the repo.

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -24,6 +24,20 @@
     <InformationalVersion Condition="'$(InformationalVersion)' == '' and '$(VersionSuffix)' == ''">$(ProductVersion)</InformationalVersion>
     <InformationalVersion Condition="'$(InformationalVersion)' == '' and '$(VersionSuffix)' != ''">$(ProductVersion)-$(VersionSuffix)</InformationalVersion>
   </PropertyGroup>
+  
+  <!-- The Default behavior in VS is to show files for the first target framework in TargetFrameworks property.
+       This is required to show all the files corresponding to all target frameworks in VS. -->
+  <ItemGroup Condition="'$(BuildingInsideVisualStudio)' == 'true' or '$(DesignTimeBuild)' == 'true'">
+    <None Include="$(MSBuildProjectDirectory)\**\*.cs"
+          Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Compile)"
```
Comment (safern) [inline Directory.Build.targets:32]: Can we remove these after all these none declarations instead of duplicating the exclude in all of them?

Comment (ViktorHofer) [reply Directory.Build.targets:32]: Adding and the removing takes longer to process. The duplication can probably be avoided otherhow.

Comment (ViktorHofer) [reply Directory.Build.targets]: fixed

**Review by safern — APPROVED** [review]

**Comment by safern** [comment]
Failure is: https://github.com/dotnet/runtime/issues/32030

**Review by Anipik — COMMENTED** [review]
lgtm

**Pull request merged**